clusterctl/clusterdeployer/bootstrap/minikube/minikube.go

```diff
@@ -28,14 +28,15 @@ import (
 type Minikube struct {
 	kubeconfigpath string
 	vmDriver       string
+    options        []string
 	// minikubeExec implemented as function variable for testing hooks
 	minikubeExec func(env []string, args ...string) (string, error)
 }
 
-func New(vmDriver string) *Minikube {
+func New(options []string) *Minikube {
```

Comment:
If these are purely optional, it would be nice to make them optional. Maybe two constructors: `New()` and `WithOptions(options []string)` would make more sense (and make the test code easier to read)?

Client code would look like either `m := minikube.New()` or `m := minikube.WithOptions([]string{"vm-driver=kvm2"})` which I think would be easier to read than `m := New([]string{})`.

Reply:
TBH, I don't mind either way so I'll go with whatever you guys reckon is best. :+1:

Reply:
Let's go with two constructors. We might only end up using the `WithOptions` one in the normal code, but it'll make the test code look a lot nicer.